rt_utils/utils.py

```diff
@@ -20,15 +46,42 @@ class ROIData:
     use_pin_hole: bool = False
 
     def __post_init__(self):
+        self.validate_color()
         self.add_default_values()
 
     def add_default_values(self):
-        if self.color == None:
-            self.color = self.get_random_colour()
+        if self.color is None:
+            self.color = COLOR_PALLET[(self.number - 1) % len(COLOR_PALLET)]
 
-        if self.name == None:
+        if self.name is None:
             self.name = f"ROI-{self.number}"
-    
-    def get_random_colour(self):
-        max = 256
-        return [randrange(max), randrange(max), randrange(max)]
\ No newline at end of file
+
+    def validate_color(self):
```

Comment:
don't need to change but we should try to keep methods a tad smaller

Reply:
#clean code book